userspace/libsinsp/cri.cpp

```diff
@@ -143,6 +143,33 @@ bool cri_interface::parse_cri_image(const runtime::v1alpha2::ContainerStatus &st
 		}
 	}
 
+	if(image_name.empty() || strncmp(image_name.c_str(), "sha256", 6) == 0)
+	{
+		/* Retrieve image_name from annotations as backup when image name may start with sha256,
+		brute force try each schema we know of for containerd and crio container runtimes. */
+
+		Json::Value root;
+		Json::Reader reader;
+		if(reader.parse(info.find("info")->second, root))
+		{
+			Json::Value jvalue;
+			jvalue = root["runtimeSpec"]["annotations"]["io.kubernetes.cri.image-name"];
+			if(jvalue.isNull())
+			{
+				jvalue = root["runtimeSpec"]["annotations"]["io.kubernetes.cri-o.Image"];
+			}
+			if(jvalue.isNull())
+			{
+				jvalue = root["runtimeSpec"]["annotations"]["io.kubernetes.cri-o.ImageName"];
+			}
```

Comment:
Same, what happens if there's no runtimeSpec or no annotations? Again, temporarily introducing typos could help :)

Reply:
JsonCPP returns null (just tested); i think the code is fine.